### PR TITLE
add support for groups extension

### PR DESCRIPTION
**Pull request by exaexa**

### CODE REVIEW

**Comment by exaexa** [comment]
(@giordano this might take some time to debug but any preliminary comments very welcome :) )

**Comment by exaexa** [comment]
note to self: TODO improve coverage with adding sbml testcases

File: src/structs.jl [hunk 1]
```diff
@@ -450,6 +450,41 @@ end
...
+    sbo::Maybe{String} = nothing
+    cv_terms::Vector{CVTerm} = []
+end
+
+"""
+$(TYPEDEF)
+
 Structure that collects the model-related data. Contains `parameters`, `units`,
 `compartments`, `species` and `reactions` and `gene_products`, and additional
 `notes` and `annotation` (also present internally in some of the data fields).
```
Comment (giordano) [inline src/structs.jl:490]: This list is getting out of date.  But at least all fields are automatically listed in the section below.  Not sure it's worth naming the fields here?

**Review by giordano — APPROVED** [review]
Looks good to me!

**Comment by exaexa** [comment]
@mihai-sysbio btw if you could have a quick look at `src/structs.jl` and confirm that it would be sufficient to support the StdGem/YeastGem usecases it would be great :] Thanks!

**Review by stelmo — APPROVED** [review]
SBML 😨

File: src/structs.jl [hunk 2]
```diff
@@ -450,6 +450,41 @@ end
 """
 $(TYPEDEF)
 
+# Fields
+$(TYPEDFIELDS)
+"""
+Base.@kwdef struct Member
```
Comment (giordano) [inline src/structs.jl:456]: Do these new structs need to be added to https://github.com/LCSB-BioCore/SBML.jl/blob/64ec00b69130a70cf75775792e76c7db6003b1c3/test/common.jl#L22-L33?

Comment (exaexa) [reply src/structs.jl:456]: good point thx, done now

File: src/structs.jl (hunk 1 above)
Comment (exaexa) [reply src/structs.jl]: solved now

**Comment by mihai-sysbio** [comment]
> @mihai-sysbio btw if you could have a quick look at `src/structs.jl` and confirm that it would be sufficient to support the StdGem/YeastGem usecases it would be great :] Thanks!

Sorry I was slow to reply. It looks like a very flexible structure is now I place 👍🏻  I must admit I haven't looked into other SBMLs and how the structure these groups, so I would have guessed the Members of a Group would have fewer attributes.

**Pull request merged**